opentelemetry-sdk/tests/trace/test_trace.py

```diff
@@ -114,6 +114,34 @@ def test_start_span_explicit(self):
             self.assertIs(tracer.get_current_span(), root)
             self.assertIsNotNone(child.end_time)
 
+    def test_span_members(self):
+        context = contextvars.ContextVar('test_span_members')
+        tracer = trace.Tracer(context)
+
+        other_neighbor = trace_api.SpanContext(
+            trace_id=0x0000000000000000000000000000cafe,
+            span_id=0x00000000cafe1234
+        )
+
+        self.assertIsNone(tracer.get_current_span())
+
+        with tracer.start_span('root') as root:
+            root.set_attribute('component', 'http')
+            root.set_attribute('http.method', 'GET')
+            root.set_attribute('http.url',
+                               'https://example.com:779/path/12/?q=d#123')
+            root.set_attribute('http.status_code', 200)
+            root.set_attribute('http.status_text', 'OK')
+
+            # Setting an attribute with the same key as an existing attribute
+            # SHOULD overwrite the existing attribute's value.
+            root.set_attribute('attr-key', 'attr-value1')
+            root.set_attribute('attr-key', 'attr-value2')
+
+            root.add_event('event1', 'event1-detail')
+
+            root.add_link(other_neighbor, 'neighbor')
```

Comment:
`add_link` requires a `SpanContext`, do we intend to revisit this later?

Reply:
I am not sure to follow: on this line, the `other_neighbor` argument is a `SpanContext`, created above:
```
        other_neighbor = trace_api.SpanContext(
            trace_id=0x0000000000000000000000000000cafe,
            span_id=0x00000000cafe1234
        )
```

Reply:
Sorry I thought it was a string. You're right.

I guess the correct invocation should be something like this? `root.add_link(other_neighbor, {'name': 'neighbor'})`
I also wonder if the API should treat `attributes` as optional? `root.add_link(my_span_context)`

Reply:
> I also wonder if the API should treat attributes as optional? 

Oh actually it should (same with `add_event()`)

Related question: should we add `Link`/`Event` classes or keep with this minimalistic approach? cc @c24t

Reply:
This PR seems like a great place to add `Link`/`Event`.

Reply:
@reyang points out that we could use a singleton default empty attribute map here (like  @carlosalberto's [`CURRENT_SPAN`](https://github.com/c24t/opentelemetry-python/blob/e8ef980daf3c72abc699a654dfe6c4e5b8bfbcda/opentelemetry-api/src/opentelemetry/trace/__init__.py#L232) to avoid creating a bunch of empty attributes.